File: GarageGo-BE/Models/Xe.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace GarageGo_BE.Models
{
    [Table("XE")]
    public class Xe
    {
        [Key]
        public int MaXe { get; set; }

        [Required]
        [StringLength(20)]
        public string BienSo { get; set; } = string.Empty; // ✅ Thêm

        [StringLength(50)]
        public string? HangXe { get; set; }

        public int? DoiXe { get; set; }

        [StringLength(30)]
        public string? MauXe { get; set; }

        [Required]
        public int MaKH { get; set; }

        [ForeignKey("MaKH")]
        public KhachHang? KhachHang { get; set; }
    }
}

File: GarageGo-BEJava/src/main/java/com/garagego/model/Xe.java
package com.garagego.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "XE")
public class Xe {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "MaXe")
    private Integer maXe;

    @Column(name = "BienSo", nullable = false, unique = true, length = 20)
    private String bienSo;

    @Column(name = "HangXe", length = 50)
    private String hangXe;

    @Column(name = "DoiXe")
    private Integer doiXe;

    @Column(name = "MauXe", length = 30)
    private String mauXe;

    @Column(name = "MaKH", nullable = false)
    private Integer maKH;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "MaKH", insertable = false, updatable = false)
    private KhachHang khachHang;
}
